lib/Transport.js

```diff
@@ -301,6 +310,12 @@ class Transport {
       response.on('end', onEnd)
       response.on('aborted', onAbort)
     }
+    // Helper function to check if memory circuit breaker enabled and the response payload is too large to fit into available heap memory.
+    const shouldApplyCircuitBreaker = (contentLength) => {
+      const heapUsed = process.memoryUsage().heapUsed
```

Comment:
Due to garbage collection can happens at any time. I think we need to the capture the real time memory usage here.

Reply:
True but I'm not sure if this possible but will contentLength ever be bigger than HEAP_SIZE_LIMIT and max heap %? If so I don't see why would need to access the heapUsed since we already know contentLength is already passed the threshold.

For example:

```
const threshold = HEAP_SIZE_LIMIT * this.responseMaxHeapPercentage;
if (contentLength > threshold) return true;
const heapUsed = process.memoryUsage().heapUsed;
return contentLength + heapUsed > threshold;
```

The first two lines should be negligible in runtime compared to when `heapUsed` is accessed. But if some application uses this circuit breaker and hits OpenSearch a lot that returns a lot of results at least we can try to capture the events where the contentLength was just way too big before we access the heap used. WDYT?

Reply:
Thanks, I see your point now.
I agree that this is nice to have but in quite extreme case. If you see the overall `contentLength` `if else`check, we start with `MAX_BUFFER_LENGTH` and `MAX_STRING_LENGTH` check (The constraint varies based on nodeJs version, lets say node 14, then `MAX_STRING_LENGTH == 512 MB`).  So the first and second checks will filter out most cases when `contentLength` is too large.  When we enter the new added `else if` block, the most affected factor in this scenario will be the `current used heap but not garbage collected yet`. That is why I think `process.memoryUsage().heapUsed` should be called in most cases. 
Your suggestion is also quite valid, I am good to update